client/src/components/Schedule/SettingsDrawer/ShowTableColumns.tsx

```diff
@@ -16,15 +17,24 @@ interface ShowTableColumnsProps {
   setColumnsHidden: (value: string[]) => void;
   eventTypesHidden: string[];
   setEventTypesHidden: (value: string[]) => void;
+  closeDrawer: () => void;
 }
 
 const ShowTableColumns: React.FC<ShowTableColumnsProps> = ({
   eventTypes,
-  columnsHidden,
-  setColumnsHidden,
-  eventTypesHidden,
-  setEventTypesHidden,
+  closeDrawer,
+  columnsHidden: initialColumnsHidden,
+  setColumnsHidden: setInitialColumnsHidden,
+  eventTypesHidden: initialEventTypesHidden,
+  setEventTypesHidden: setInitialEventTypesHidden,
 }) => {
+  const [columnsHidden, setColumnsHidden] = useState<string[]>(initialColumnsHidden);
+  const [eventTypesHidden, setEventTypesHidden] = useState<string[]>(initialEventTypesHidden);
+
+  const isResetButtonDisabled =
+    isEqual([...columnsHidden].sort(), [...initialColumnsHidden].sort()) &&
+    isEqual([...eventTypesHidden].sort(), [...initialEventTypesHidden].sort());
```

Comment:
does it make sense to memo this op? 4 sorts, 4 new arrays on each render

Reply:
You're right. Fixed